dds_registration/models.py

```diff
@@ -285,29 +290,55 @@ def invoice_pdf(self):
+            email_template = "dds_registration/payment/emails/invoice_membership.txt.django"
         else:
-            event = Event.objects.get(id=self.data["event"]["id"])
-            subject = f"DdS Event {event.title} Registration Invoice {self.invoice_no}"
-            message = f"Thanks for registering for {event.title}! We look forward to seeing your, in person or virtually.\n\nDépart de Sentier runs its events and schools on a cost-neutral basis - i.e. we don't make a profit off the registration fees. They are used for catering, room, hotel, and equipment rental, AV hosting and technician fees, and guest speaker costs. We literally could not run this event without your support.\n\nYou can view your registration status and apply for membership at https://events.d-d-s.ch/profile.\n\nPlease find attached the registration invoice. Your registration is not finalized until the bank transfer is received.\n\nYou can change your invoice details here: https://events.d-d-s.ch{reverse('event_registration', args=(event.code,))}.\n\nIf you have any questions, please contact events@d-d-s.ch."
+            context["event"] = Event.objects.get(id=self.data["event"]["id"])
+            email_template = "dds_registration/payment/emails/invoice_membership.txt.django"
+        # Parse email message template
+        text = render_to_string(
+            template_name=email_template,
+            context=context,
+            request=request,
+        )
+        # Extract a subject and a message from the template
+        [subject, message] = parse_email_subject_and_content(text)
         user.email_user(
             subject=subject,
             message=message,
             attachment_content=self.invoice_pdf(),
             attachment_name=f"DdS Invoice {self.invoice_no}.pdf",
         )
 
-    def email_receipt(self):
+    def email_receipt(self, request: HttpRequest):
         user = User.objects.get(id=self.data["user"]["id"])
-        kind = "Membership" if self.data["kind"] == "membership" else "Event"
+        context = {
+            "kind": "Membership" if self.data["kind"] == "membership" else "Event",
+            "site": Site.objects.get_current(),
+            "scheme": "https" if request.is_secure() else "http",
+            "payment": self,
+            "user": user,
+        }
+        email_template = "dds_registration/payment/emails/receipt.txt.django"
+        # Parse email message template
+        text = render_to_string(
+            template_name=email_template,
+            context=context,
+            request=request,
```

Comment:
You had to make a lot of changes elsewhere to get the request here, but I don't understand why it is needed.